### PR TITLE
plasma5: 5.20 -> 5.21

### PR DESCRIPTION
**Pull request by CertainLach**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by CertainLach** [comment]
Plasma 5.21 is not released yet, but it improves wayland support, so i testing it on nixos and upgrading patches, to make it released on NixOS in same time as in other distros :D 

I am not sure if i should split kdeFrameworks upgrade to a different pull request, because plasma 5.20 seems to not support frameworks 5.76

File: pkgs/desktops/plasma-5/ksysguard.nix [hunk 1]
```diff
@@ -12,6 +13,6 @@ mkDerivation {
   nativeBuildInputs = [ extra-cmake-modules kdoctools ];
   buildInputs = [
     kconfig kcoreaddons kitemviews knewstuff kiconthemes libksysguard
-    kdelibs4support ki18n libcap libpcap lm_sensors
+    kdelibs4support ki18n libcap libpcap lm_sensors kdeFrameworks.networkmanager-qt libnl
```
Comment (SuperSandro2000) [inline pkgs/desktops/plasma-5/ksysguard.nix:16]: Generally we do not want to pass in one level lower than the package. Not sure if this is a common pattern in kde though.

File: pkgs/desktops/plasma-5/plasma-browser-integration.nix [hunk 2]
```diff
@@ -3,7 +3,7 @@
   extra-cmake-modules,
   qtbase,
   kfilemetadata, kio, ki18n, kconfig , kdbusaddons, knotifications, kpurpose,
-  krunner, kwindowsystem, kactivities,
+  krunner, kwindowsystem, kactivities, plasma-workspace,
```
Comment (SuperSandro2000) [inline pkgs/desktops/plasma-5/plasma-browser-integration.nix:6]:
```suggestion
  krunner, kwindowsystem, kactivities, plasma-workspace
```

File: pkgs/desktops/plasma-5/ksysguard.nix (hunk 1 above)
Comment (ttuegel) [reply pkgs/desktops/plasma-5/ksysguard.nix:16]: Yes, you MUST NOT pass in kdeFrameworks. Please refer to the Nixpkgs Manual for explanation.

File: pkgs/development/libraries/kde-frameworks/fetch.sh [hunk 3]
```diff
@@ -1 +1 @@
-WGET_ARGS=( https://download.kde.org/stable/frameworks/5.76/ -A '*.tar.xz' )
+WGET_ARGS=( https://download.kde.org/stable/frameworks/5.78/ -A '*.tar.xz' )
```
Comment (eliasp) [inline pkgs/development/libraries/kde-frameworks/fetch.sh:1]: Frameworks 5.79.0 was [just released](https://kde.org/announcements/frameworks/5/5.79.0/) and for kdnssd and syntax-highlighting a 5.79.1 release did follow immediately with hotfixes.

**Comment by eddyb** [comment]
Can this wait for 5.21.1, or at least backport https://invent.kde.org/plasma/plasma-desktop/-/merge_requests/353?
It's a fix for https://bugs.kde.org/show_bug.cgi?id=431923 (`KeyboardRepeating` migration to `KeyRepeat` incorrectly disabled key repeating when it was previously enabled).

Arch users hit already, and IIUC once the incorrect config upgrade runs, there's not enough information left around to undo it (so every user would have to go to System Settings manually and re-enable key repeating).

I'm not aware of other bugs but it might be a good idea to do a quick check for anything recently reported for 5.21.0

**Comment by eliasp** [comment]
After resolving the conflicts, this will make `kcompletion-5.79` build again...
```diff
diff --git a/pkgs/development/libraries/kde-frameworks/kcompletion.nix b/pkgs/development/libraries/kde-frameworks/kcompletion.nix
index 26e5a83dc21..4813b06deb1 100644
--- a/pkgs/development/libraries/kde-frameworks/kcompletion.nix
+++ b/pkgs/development/libraries/kde-frameworks/kcompletion.nix
@@ -7,13 +7,6 @@
 mkDerivation {
   name = "kcompletion";
   meta = { maintainers = [ lib.maintainers.ttuegel ]; };
-  patches = [
-    # https://mail.kde.org/pipermail/distributions/2021-January/000928.html
-    (fetchpatch {
-      url = "https://invent.kde.org/frameworks/kcompletion/commit/7acda936f06193e9fc85ae5cf9ccc8d65971f657.patch";
-      sha256 = "150ff506rhr5pin5363ks222vhv8qd77y5s5nyylcbdjry3ljd3n";
-    })
-  ];
   nativeBuildInputs = [ extra-cmake-modules ];
   buildInputs = [ kconfig kwidgetsaddons qttools ];
   propagatedBuildInputs = [ qtbase ];
```

**Comment by eliasp** [comment]
Feel free to pull some updates for this PR from https://github.com/eliasp/nixpkgs/tree/plasma-5.21:

- bc7aecb0e393007c372aceb4582b8a190c4dd272 - kcompletion: drop obsolete patch 7acda936
- 2d28e7d7ad3909d83003119ee9a6be60537ce6eb - kio: drop obsolete patch a183dd0d
- ab2c0219d19edfac2c02fa0e814cb4b3790000c7 - plasma-desktop: add patch for bug#431923

**Comment by ttuegel** [comment]
> Can this wait for 5.21.1, or at least backport 

I don't see any harm waiting until Tuesday (when 5.21.1 is released).

**Comment by CertainLach** [comment]
Rebuilding and testing 5.21.1

**Comment by CertainLach** [comment]
Rebuilt system, everything works as intended for me

**Comment by afontaine** [comment]
My attempts to run `nix-review` on this PR have failed, all with errors roughly following the below:

```
error: --- Error ------------------------------------------------------------------------------------------------------------- nix
builder for '/nix/store/54qs81vr7q2nq18k8asv23yp1m279gr8-env.drv' failed with exit code 1; last 3 log lines:
  Error: detected mismatched Qt dependencies:
      /nix/store/jx9niyr7kv0yplsvc8iq43wbqnpcjnqw-qtbase-5.14.2-dev
      /nix/store/gwpcrjjrw7kk82hin81jhf7xv97qqh4m-qtbase-5.12.10-dev
```

Is there anything additional required here that I am missing?

**Comment by ttuegel** [comment]
> Is there anything additional required here that I am missing?

This means that there is a package outside Plasma, that depends on Plasma, that was horribly broken before this pull request.

**Review by ttuegel — CHANGES_REQUESTED** [review]
Updating KDE Frameworks makes this a mass rebuild, so it must be rebased onto the staging branch. (I think that the update is already in that branch.)

**Comment by CertainLach** [comment]
I messed up. Can i revert all of those reviewers added? :D

**Comment by mohe2015** [comment]
@CertainLach How were you able to build this? I currently get a likely unrelated build error at libe-book-0.1.3

**Comment by CertainLach** [comment]
@mohe2015 i didn't ran review on latest commit and staging rebase, but i had some errors related to failing python packages tests before.
For latest commit after rebase i successfully ran plasma test, nothing more currently.

**Comment by mohe2015** [comment]
I rebased to nixos-unstable and it seems to be building (takes some time of course). Rebasing on top of nixos-unstable should be fine or is there something required from staging?

**Comment by CertainLach** [comment]
I don't know, mine laptop is running latest commit rebased on master.
As i can see, unstable already includes needed kdeFrameworks version, so it should work.

**Comment by mohe2015** [comment]
signal-2021-02-26-175908
any idea? I rebased onto master and also added # 100057 (no link to prevent unnecessary notifications).
This is wayland and I use two monitors so I don't expect this to get fixed but would still be nice if it's an easy fix. Also the other PR alone logged me in but created strange graphic artifacts that are supposed to be fixed by this plasma update.

**Review by mohe2015 — APPROVED** [review]
Behaviour-wise LGTM! (rebased onto master).

Wayland had some issues with multiple screens but I'm pretty sure this is an upstream thing.

**Review by ttuegel — APPROVED** [review]

**Comment by ttuegel** [comment]
Thanks!

**Comment by payasrelekar** [comment]
> Behaviour-wise LGTM! (rebased onto master).
> 
> Wayland had some issues with multiple screens but I'm pretty sure this is an upstream thing.

Can you please tell me how I can use wayland instead of X11? By default it started on X and there is no session option on login to select Wayland.

**Comment by mohe2015** [comment]
@payasrelekar I think I used https://github.com/NixOS/nixpkgs/pull/100057 but I'm not sure.

**Pull request merged**